From 316a9417cba29d34f38ee71009dec219a3dc4b5f Mon Sep 17 00:00:00 2001
From: Matt Massie <matt@cloudera.com>
Date: Fri, 31 Jul 2009 18:45:00 -0700
Subject: [PATCH] Fix a bug in the hadoop/sqoop wrapper generation

---
 cloudera/install_hadoop.sh |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/cloudera/install_hadoop.sh b/cloudera/install_hadoop.sh
index afc5090..93a748d 100755
--- a/cloudera/install_hadoop.sh
+++ b/cloudera/install_hadoop.sh
@@ -121,8 +121,8 @@ done
 # Make bin wrappers
 mkdir -p $BIN_DIR
 
-for bin_wrapper in hadoop-$APACHE_BRANCH sqoop-$APACHE_BRANCH ; do
-  cat > $BIN_DIR/$bin_wrapper <<EOF
+for bin_wrapper in hadoop sqoop ; do
+  cat > $BIN_DIR/$bin_wrapper-$APACHE_BRANCH <<EOF
 #!/bin/sh
 
 export HADOOP_HOME=$INSTALLED_LIB_DIR
-- 
1.6.0.4

