From a367eec549f4a76ff042ac0b3a97684cda038c63 Mon Sep 17 00:00:00 2001
From: Todd Lipcon <todd@cloudera.com>
Date: Tue, 22 Sep 2009 22:19:02 -0700
Subject: [PATCH] Recreate c++ configure files during build if we have the right build dependencies

---
 cloudera/do-release-build |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/cloudera/do-release-build b/cloudera/do-release-build
index 90ee287..d039fe5 100755
--- a/cloudera/do-release-build
+++ b/cloudera/do-release-build
@@ -47,6 +47,18 @@ RELEASE_DIR=$BIN_DIR/..
 
 cd $RELEASE_DIR
 
+if which aclocal-1.9 && which autoconf ; then
+  ant -Dcompile.c++=true \
+    -Dcompile.native=true create-c++-configure
+fi
+
+if which libtoolize ; then
+  for dir in src/native src/examples/pipes src/c++/libhdfs/ src/c++/pipes src/c++/utils ; do
+    (cd $dir && libtoolize --force)
+  done
+fi
+
+
 JAVA_HOME=$JAVA32_HOME \
   CFLAGS=-m32 \
   CXXFLAGS=-m32 \
-- 
1.6.0.4

