From 31a79c62df9bcf785290c1000e1d61b9a0caf0f1 Mon Sep 17 00:00:00 2001
From: Chad Metcalf <chad@cloudera.com>
Date: Mon, 26 Oct 2009 10:56:44 -0700
Subject: [PATCH] Copy the scoop man page to the expected version number.

---
 cloudera/do-release-build |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/cloudera/do-release-build b/cloudera/do-release-build
index 217c257..70c156b 100755
--- a/cloudera/do-release-build
+++ b/cloudera/do-release-build
@@ -97,6 +97,8 @@ popd
 mkdir -p build/hadoop-$FULL_VERSION/docs/
 cp -r build/contrib/sqoop/doc/ build/hadoop-$FULL_VERSION/docs/sqoop
 
+cp build/hadoop-$FULL_VERSION/docs/sqoop/sqoop.1.gz \
+   build/hadoop-$FULL_VERSION/docs/sqoop/sqoop-0.20.1.gz
 
 JAVA_HOME=$JAVA64_HOME \
   CFLAGS=-m64 \
-- 
1.6.0.4

